sdk/go/common/workspace/plugins.go

```diff
@@ -353,14 +358,54 @@ func newGithubSource(url *url.URL, name string, kind PluginKind) (*githubSource,
 }
 
 func (source *githubSource) newHTTPRequest(url, accept string) (*http.Request, error) {
-	req, err := buildHTTPRequest(url, fmt.Sprintf("token %s", source.token))
+	var token string
+	if source.token != "" {
+		token = fmt.Sprintf("token %s", source.token)
+	}
+	req, err := buildHTTPRequest(url, token)
 	if err != nil {
 		return nil, err
 	}
 	req.Header.Set("Accept", accept)
 	return req, nil
 }
 
+func (source *githubSource) retryHTTP(
+	getHTTPResponse func(*http.Request) (io.ReadCloser, int64, error),
+	req *http.Request) (io.ReadCloser, int64, error) {
+
+	for attempt := 0; ; attempt++ {
+		resp, length, err := getHTTPResponse(req)
+		if err == nil {
+			return resp, length, nil
+		}
+
+		// Only retry a few times.
+		const maxAttempts = 5
+		if attempt >= maxAttempts {
+			return nil, -1, err
+		}
+
+		// We'll retry only 403 rate limit errors.
+		var downErr *downloadError
+		if !errors.As(err, &downErr) {
```

Comment:
Did we need to still do a status code check here?
```suggestion
		// We'll retry only 403 rate limit errors.
		var downErr *downloadError
		if !errors.As(err, &downErr) || downErr.code == 403 {
```